### DISCUSSION TITLE
Google Lighthouse 💯

### DISCUSSION THREAD

**Discussion by cybacoffee** [lucky-media/cloud#6] [Ideas]
Hi guys 
I love the design, but would it be possible to work on all Google Lighthouse 100 scores (https://web.dev)? Would be very great to achieve that, then I would definitely buy it! 😃

Thanks and keep up the awesome work!

**Comment by lokmanm** [comment]
Hi,

Thanks a lot for bringing this up. In the next release of cloud we have planned to add a pricing section which will be released the following days.

After that release our focus will be to fix some of the issues with Google Lighthouse. 

I will post an update once we have it fixed.

**Comment by lokmanm** [comment]
Hey @cybacoffee, with the latest release of [v1.3](https://github.com/lucky-media/cloud/releases/tag/v1.3) we have switched to use the responsive image partial for all the images in the starter kit. This should fix all the issues related to Google Lighthouse. 

Please note, we haven't updated our demo website yet but we plan to in the following days.

Thanks again for the notice.

**Reply by andylarkum** [reply]
On this - I'm dropping performance score because imaged (specifically SVGs) are served with no explicit width/height. 
Any idea how to solve that?!

**Reply by lokmanm** [reply]
> 

Hey @andylarkum, did you use an `svg` inside an image ? Can you give me more info on this so I can look it up ? 

Info so I can help you, send me the partial where you uploaded the image so I can check.

If the partial was expecting an image and you uploaded an `svg` by default it will resort to the `<img>` tag so It can render the svg correctly.

**Reply by andylarkum** [reply]
Hi @lokmanm 

Yuh, so it's on any hero image where using an svg. It does resort to the <img> tag, as you say. I can't find anything that explains how to set a width/height for an svg, and it's hurting the CLS score. 

Any idea how to set a responsive width/height on an svg to satisfy the almighty Google?!

Thanks!
A.

**Reply by lokmanm** [reply]
@andylarkum one way to resolve this is to render the `svg` directly using the `{{ svg }}` tag.

One way to do this is to edit the `picture` partial like so ( I haven't tested this if it will break anything ).

This is how it looks now ( part of the code )

```
{{ if extension == 'svg' || extension == 'gif' }}
   <img class="{{ class }}" src="{{ url }}" alt="{{ alt }}" />
{{ else }}
```

This can be modified to this:

```
{{ if extension == 'svg' }}
   {{ svg :src="url" :class="class" }}
{{ elseif extension == 'gif'  }}
  <img class="{{ class }}" src="{{ url }}" alt="{{ alt }}" >
```

rest of the code is not copy pasted but I hope you get the gist.

**Reply by andylarkum** [reply]
Ah! Should have thought of that! Thank you!

I also added:
```
width="{{ width }}" 
height="{{ height }}" 
```
...to the img type in that component (it's in resources > views > components > picture.antlers.html in case anyone else needs this!)

**Reply by lokmanm** [reply]
@andylarkum awesome, glad I could help !

**Reply by andylarkum** [reply]
Hmmm, shux, I should have tested more - it breaks other pictures on the page. i.e. it fixes the hero image, but in a "section with image", the svg fills the page with text over the top. :(
